rpc/core/routes.go

```diff
@@ -37,25 +40,58 @@ func (env *Environment) GetRoutes() RoutesMap {
 		"consensus_params":     rpc.NewRPCFunc(env.ConsensusParams, "height", true),
 		"unconfirmed_txs":      rpc.NewRPCFunc(env.UnconfirmedTxs, "limit", false),
 		"num_unconfirmed_txs":  rpc.NewRPCFunc(env.NumUnconfirmedTxs, "", false),
+	}
```

Comment:
Some of these routes may rely on uninitialized data when using in debug mode so these might need to be further split down.

Reply:
Yup for sure. There are endpoints here that are reliant on the mempool and the consensus engine. I'm definitely for categorising these different endpoints. If you look at `rpc/core/interface.go` you will see a similar pattern. We might want to consider dividing not based on request type but on the resources necessary. This is something that maybe the others want to weigh in on. In any case, if we are going to have partially filled `Environment`, it needs to be able to correctly handle all situations where it gets a request and for one reason or another doesn't have the resources to fulfill that request.

Reply:
I'll take a crack at fixing this up. I would add that I think these APIs should be separated by utility instead of by requirement. There is inevitably going to be some overlap in requirements: some will take a mempool, some will take a state store, some will take both etc. So dividing in that way will either require doing all sub-combinations or similar?

Reply:
I feel this may be another argument in favour of having the Debug node live in a separate package, and only export the RPC surfaces it can safely support. 

Or even if the debug simulation needs to export the others to satisfy the interface, it can do so with stubs that report sensible errors, rather than trying to grobble into uninitalized or unknown state, potentially worsening the confusion for the debugger. 🙂

Reply:
> I feel this may be another argument in favour of having the Debug node live in a separate package, and only export the RPC surfaces it can safely support.

@creachadair mind elaborating on this a bit. Is your proposal here to have a separate debug package _within_ the rpc package or within the _node_ package? I think having it be within the _rpc_ package may make sense. This would mean that the `rpc/debug` package will have constructors that take all of the types that they need to construct the endpoints for the debug endpoints to make the dependencies clear while also keeping the RPC code/types somewhat self contained. Thoughts?

Reply:
I was thinking in terms of a package entirely separate from both: One that imports both `node` and `rpc` separately maybe. I didn't really think carefully about having a subpackage of `node` or `rpc` themselves. My main hope: The caller should be able to choose whether or not to import the debug functionality, rather than having it come in by default.

Your reasoning about having it be part of `rpc` makes sense to me, and I don't think it's necessarily bad to put it there. The only question I think is whether that makes the dependency hairball worse. Ideally neither `node` nor `rpc` should have to automatically depend on the other directly. I think right now there is some entanglement—but we should probably not make it worse if we can help it. 

Does that help clarify at all?

Reply:
Moved all rpc endpoints relevant to this new functionality into an `inspect/rpc` package. This is similar to how the `light` package works.